### PR TITLE
Narrow the scope of the Windows Android arm hack

### PR DESCRIPTION
**Pull request by zanderso**
Engine side of https://github.com/flutter/buildroot/pull/645

### CODE REVIEW

**Comment by zanderso** [comment]
Presubmits are angry at the python script formatting. I'll fix it before landing.

File: DEPS [hunk 1]
```diff
@@ -118,7 +118,7 @@ allowed_hosts = [
 ]
 
 deps = {
-  'src': 'https://github.com/flutter/buildroot.git' + '@' + 'ad96f00615dc306c22cc09d433eee50392f8033c',
+  'src': 'https://github.com/zanderso/buildroot.git' + '@' + '37f0ef84bfd2121d126cd6226d440e8228a79a1a',
```
Comment (godofredoc) [inline DEPS:121]: Will this be updated after https://github.com/flutter/buildroot/pull/645 is landed?

**Review by godofredoc — APPROVED** [review]
LGTM after updating the DEPS file with https://github.com/flutter/buildroot/pull/645

File: tools/gn [hunk 2]
```diff
@@ -325,15 +325,16 @@ def to_gn_args(args):
...
-  if sys.platform.startswith(('cygwin', 'win')) and args.target_os != 'win':
-    cpu = cpu_for_target_arch(gn_args['target_cpu'])
-    # We explicitly allow arm64 native build. 'host_cpu' key may not exist.
-    if gn_args.get('host_cpu') == 'arm64' and gn_args['target_cpu'] == 'arm64':
-      cpu = 'arm64'
-    gn_args['host_cpu'] = cpu
-    gn_args['target_cpu'] = cpu
+  # We cannot cross-compile for 32 bit arm on a Windows host. We work around
+  # this by leaving 'target_cpu' and 'dart_target_arch' set to 'arm' so that
+  # Dart tools like gen_snapshot that are built for the host will correctly
```
Comment (cbracken) [inline tools/gn:330]:
```suggestion
  # Dart tools such as gen_snapshot that are built for the host will correctly
```

**Review by cbracken — APPROVED** [review]
LGTM modulo switching back to flutter/buildroot from zanderso/buildroot as noted by @godofredoc.

File: DEPS (hunk 1 above)
Comment (zanderso) [reply DEPS]: Done.

File: tools/gn (hunk 2 above)
Comment (zanderso) [reply tools/gn]: Done.

**Pull request merged**